src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -405,6 +408,22 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
     addr
   }
 
+  def NewSharedArray[T:Manifest](x: Rep[Int]): Rep[Array[T]] = {
+    Wrap[Array[T]](Adapter.g.reflectMutable("NewSharedArray", Unwrap(x)))
+  }
+
+  def NewSharedArray[T:Manifest](x: Rep[Int], y: Rep[Int]): Rep[Array[Array[T]]] = {
+    Wrap[Array[Array[T]]](Adapter.g.reflectMutable("NewSharedArray", Unwrap(x), Unwrap(y)))
+  }
+
+  def NewSharedArray[T:Manifest](x: Rep[Int], y: Rep[Int], z: Rep[Int]): Rep[Array[Array[Array[T]]]] = {
+    Wrap[Array[Array[Array[T]]]](Adapter.g.reflectMutable("NewSharedArray", Unwrap(x), Unwrap(y), Unwrap(z)))
+  }
```

Comment:
Just gave more thought to it and I think all three functions here should take `Int` typed parameter instead of `Rep[Int]`, since the parameter should be some const value in generated code. Do you agree?